coredocument/coredocument.proto

```diff
@@ -78,6 +78,25 @@ message CoreDocument {
   ];
   // read rules
   repeated ReadRule read_rules = 19;
+  // AccessTokens which have been added to this CoreDocument
+  repeated AccessToken access_token = 20;
```

Comment:
Oh wait, why don't we add the access token directly in the roles?

Reply:
ah, I thought just the list of identifiers would be in the roles, but actually I think you are right.

Reply:
I guess it actually depends on how precise-proofs handles this, whether or not the list of identifiers that then act as pointers to specific access tokens makes more sense than having the repeated entries of access tokens themselves.

Reply:
I guess with the way that precise-proofs works, it would make sense to have the access tokens directly listed in the roles, that way they are picked as leaves on the resulting merkle proof tree. The identifiers way seems redundant.

Reply:
I believe logically an access token is a role same as nfts, collabs... So it makes sense to have it within the role struct. If we need to know simply if a identifier exists in the list of access tokens of that doc then I would add something like:
```
repeated AccessToken access_token = 20 [
    (proofs.key_length) = 32,
    (proofs.mapping_key) = "token_key"
];
```
And then we cover both cases.

Reply:
okay, yes, makes sense.

Reply:
cool, I meant actually:
```
(proofs.key_length) = 32,
(proofs.mapping_key) = "identifier"
```

Reply:
yea, I recognised your snippet from Roles :)